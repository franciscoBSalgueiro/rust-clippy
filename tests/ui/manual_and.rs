#![warn(clippy::manual_and)]
#[allow(irrefutable_let_patterns)]

fn main() {
    let a = true;
    let b = false;
    let c = true;

    let _ = if a { b } else { false };
    //~^ ERROR: this `if`-then-`else` expression can be simplified with `&&`

    let _ = if a && c { b } else { false };
    //~^ ERROR: this `if`-then-`else` expression can be simplified with `&&`

    let _ = if a || c { b } else { false };
    //~^ ERROR: this `if`-then-`else` expression can be simplified with `&&`

    // Should not lint

    // with if-let
    let _ = if let x = a { x } else { false };

    // when the then-branch is a block
    let _ = if let x = a {
        println!("hi");
        x
    } else {
        false
    };

    // when part of a chain of if-elses
    let _ = if a {
        b
    } else if b {
        a
    } else {
        false
    };
}
