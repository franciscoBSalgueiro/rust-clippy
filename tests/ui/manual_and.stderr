error: this `if`-then-`else` expression can be simplified with `&&`
  --> tests/ui/manual_and.rs:9:13
   |
LL |     let _ = if a { b } else { false };
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `a && b`
   |
   = note: `-D clippy::manual-and` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::manual_and)]`

error: this `if`-then-`else` expression can be simplified with `&&`
  --> tests/ui/manual_and.rs:12:13
   |
LL |     let _ = if a && c { b } else { false };
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `a && c && b`

error: this `if`-then-`else` expression can be simplified with `&&`
  --> tests/ui/manual_and.rs:15:13
   |
LL |     let _ = if a || c { b } else { false };
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `a || c && b`

error: aborting due to 3 previous errors

